src/compiler/transformers/declarations/emitBinder.ts

```diff
@@ -0,0 +1,914 @@
+                bindTypeParameters(interfaceDeclaration.typeParameters);
+            });
+            withMembers(interfaceSymbol, "members", () => {
+                interfaceDeclaration.members.forEach(m => {
+                    bindNode(m);
+                });
+            });
+        }
+        function bindClassDeclaration(node: ClassExpression | ClassDeclaration) {
+            const isExported = hasSyntacticModifier(node, ModifierFlags.Export);
+            const classDeclaration = node;
+            const classSymbol = addLocalAndExportDeclaration(getStatementName(classDeclaration), classDeclaration, getSymbolFlagsForNode(classDeclaration), isExported);
+            withScope(classDeclaration, /*exports*/ undefined, () => {
+                bindTypeParameters(classDeclaration.typeParameters);
+            });
+            withMembers(classSymbol, "members", () => {
+                classDeclaration.members.forEach(m => {
+                    if (hasSyntacticModifier(m, ModifierFlags.Static)) return;
+                    if (m.kind === SyntaxKind.SemicolonClassElement || m.kind === SyntaxKind.ClassStaticBlockDeclaration) return;
+
+                    bindNode(m);
+                });
+            });
+            withMembers(classSymbol, "exports", () => {
+                classDeclaration.members.forEach(m => {
+                    if (!hasSyntacticModifier(m, ModifierFlags.Static)) return;
+                    if (
+                        m.kind === SyntaxKind.SemicolonClassElement
+                        || m.kind === SyntaxKind.ClassStaticBlockDeclaration
+                    ) return;
+
+                    bindNode(m);
+                });
+            });
+        }
+        function bindDeclaration(node: Declaration) {
+            const flags = tryGetSymbolFlagsForNode(node);
+            // Unsupported declaration
+            if (!flags) {
+                return;
+            }
+            addLocalOnlyDeclaration(getMemberKeyFromElement(node), node, flags);
+            bindChildren(node);
+        }
+    }
+}
+
+/** @internal */
+export type MemberKey = string & {
+    __memberKey: void;
+};
+
+/**
+ * Gets the symbolic name for a member from its type.
+ * @internal
+ */
+export function getMemberKey(name: string | Exclude<PropertyName, ComputedPropertyName> | NoSubstitutionTemplateLiteral): MemberKey;
+/**
+ * Gets the symbolic name for a member from its type.
+ * @internal
+ */
+export function getMemberKey(name: string | PropertyName | NoSubstitutionTemplateLiteral | undefined): MemberKey | undefined;
+export function getMemberKey(name: string | PropertyName | NoSubstitutionTemplateLiteral | undefined): string | undefined {
+    if (name === undefined) {
+        return undefined;
+    }
+    if (typeof name === "string") {
+        return ("I:" + name);
+    }
+    if (isPrivateIdentifier(name)) {
+        return ("P:" + name.escapedText);

```

Comment:
The basic idea is that all names get a prefixed: 

- `I:` is for names that are reliably known (identifiers, strings, numbers). `I` is used even if we have a computed property, as long as it is a string or a number
- `C:` is for computed properties that use identifiers or dotted identifiers. 
- `P:` is for private identifiers.

I must admit that identifiers are mostly added for completeness. Since these identifiers don't really make it into declarations they will not be looked up.

The reason for using a different system is mostly due to the fact that we need a way to represent computed properties with identifiers and dotted identifiers, a requirement the regular checker/binder does not have, since it can late bind those members when type info is available.